homeassistant/components/librehardwaremonitor/coordinator.py

```diff
@@ -0,0 +1,108 @@
+"""Coordinator for LibreHardwareMonitor integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from librehardwaremonitor_api import (
+    LibreHardwareMonitorClient,
+    LibreHardwareMonitorConnectionError,
+    LibreHardwareMonitorNoDevicesError,
+)
+from librehardwaremonitor_api.model import LibreHardwareMonitorData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+type LibreHardwareMonitorConfigEntry = ConfigEntry[LibreHardwareMonitorCoordinator]
+
+
+class LibreHardwareMonitorCoordinator(DataUpdateCoordinator[LibreHardwareMonitorData]):
+    """Class to manage fetching LibreHardwareMonitor data."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            config_entry=config_entry,
+            update_interval=timedelta(seconds=DEFAULT_SCAN_INTERVAL),
+        )
+
+        host = config_entry.data[CONF_HOST]
+        port = config_entry.data[CONF_PORT]
+        self._api = LibreHardwareMonitorClient(host, port)
+        device_entries = dr.async_entries_for_config_entry(
+            registry=dr.async_get(self.hass), config_entry_id=config_entry.entry_id
+        )
+        self._previous_devices = {
+            device.name for device in device_entries if device.name is not None
+        }
```

Comment:
I would recommend to split off the stale and dynamic devices for now

Reply:
I understand the scope of the PR has to be reasonably small, that why I removed the reconfigure flow, the device selection and more.
In this case, I'm hesitant to remove it. Having the devices in HA reflect the actual devices exposed by LHM is a very nice feature to have and a real user-facing benefit. This functionality is of course covered by unit tests.

Reply:
Oh I agree that its a nice feature, but I think we can be a bit quicker with reviewing without this, and then we can add this in a follow-up

Reply:
I'd really like to keep it, hope it doesn't add too much review effort. As there is no urgency for this PR I think we can afford to keep it and avoid device inconsistencies in users' configs that we probably will have to clean up later.